Premiumizer.py

```diff
@@ -265,91 +277,107 @@ def clean_name(original):
+            eta = ' ' + utils.time_human(tmp, fmt_short=True)
+        task.update(speed=(utils.sizeof_human(speed)+ '/s'), progress=progress, eta=eta)
+
+    elif downloader.get_status() == 'combining':
+        task.update(speed='',eta=' Combining files')
+    elif downloader.get_status() == 'paused':
+        task.update(speed='',eta=' Download paused')
     else:
-        logger.debug('Want to update stats, but downloader does not exist yet.')
+        logger.debug('Want to update stats, but downloader status is invalid.')
 
 
-def download_file(task, full_path, url):
+def download_file(download_list):
     logger.debug('def download_file started')
-    logger.info('Downloading file: %s', full_path)
-    if not os.path.isfile(full_path):
-        global downloader
-        downloader = SmartDL(url, full_path, progress_bar=False, logger=logger)
-        stat_job = scheduler.scheduler.add_job(get_download_stats, args=(task,), trigger='interval', seconds=1,
-                                               max_instances=1, next_run_time=datetime.datetime.now())
-        downloader.start(blocking=True)
-        while not downloader.isFinished():
-            print('wrong! waiting for downloader before finished')
-        stat_job.remove()
-        if downloader.isSuccessful():
-            global total_size_downloaded
-            total_size_downloaded += downloader.get_dl_size()
-            logger.info('Finished downloading file: %s', full_path)
+    total_size_downloaded = 0
+    dltime = 0
+    for download in download_list:
+        logger.debug('Downloading file: %s', download['path'])
+        if not os.path.isfile(download['path']):
+            downloader = SmartDL(download['url'], download['path'], progress_bar=False, logger=logger)
+            downloader.start(blocking=False)
+            while not downloader.isFinished():
+                get_download_stats(download['task'], downloader, total_size_downloaded)
+                gevent.sleep(2)
+            if downloader.isSuccessful():
+                dltime += downloader.get_dl_time()
+                total_size_downloaded += downloader.get_dl_size()
+                logger.debug('Finished downloading file: %s', download['path'])
+                download['task'].update(dltime=dltime)
+            else:
+                logger.error('Error while downloading file from: %s', download['path'])
+                for e in downloader.get_errors():
+                    logger.error(str(e))
         else:
-            logger.error('Error while downloading file from: %s', full_path)
-            for e in downloader.get_errors():
-                logger.error(str(e))
-    else:
-        logger.info('File not downloaded it already exists at: %s', full_path)
+            logger.info('File not downloaded it already exists at: %s', download['path'])
+
 
 
 # TODO continue log statements
 
-def process_dir(task, path, dir_content):
+def process_dir(task, path, dir_content,change_dldir):
```

Comment:
No, it's for:
`            if change_dldir:
                task.update(dldir=new_path)`
this will run the first time only to update the path(dldir) to where the download is saved in instead of the category download loction